services/web/client/source/class/osparc/dashboard/StudyBrowserButtonItem.js

```diff
@@ -224,21 +224,21 @@ qx.Class.define("osparc.dashboard.StudyBrowserButtonItem", {
       let accessRights = {};
       switch (studyData["resourceType"]) {
         case "study":
-          uuid = studyData.uuid;
-          owner = studyData.prjOwner;
-          accessRights = studyData.accessRights;
+          uuid = studyData.uuid ? studyData.uuid : uuid;
+          owner = studyData.prjOwner ? studyData.prjOwner : owner;
+          accessRights = studyData.accessRights ? studyData.accessRights : accessRights;
           defaultThumbnail = this.self().STUDY_ICON;
           break;
         case "template":
-          uuid = studyData.uuid;
-          owner = studyData.prjOwner;
-          accessRights = studyData.accessRights;
+          uuid = studyData.uuid ? studyData.uuid : uuid;
+          owner = studyData.prjOwner ? studyData.prjOwner : owner;
+          accessRights = studyData.accessRights ? studyData.accessRights : accessRights;
           defaultThumbnail = this.self().TEMPLATE_ICON;
           break;
         case "service":
-          uuid = studyData.key;
-          owner = studyData.owner;
-          accessRights = studyData.access_rights;
+          uuid = studyData.key ? studyData.key : uuid;
```

Comment:
not sure what the key is supposed to be but you probably know what you're doing =P

Reply:
In the case of studies and templates, the uuid field is the unique identifier. On the other hand, the key (together with the version) makes a service unique.